### PR TITLE
Support CREATE EXTENSION, take two.

### PR DESCRIPTION
**Pull request by haradh1**
Instead of the previous attempt to support this in madpack time, this time
we try to make the build system to create EXTENSION files so that upcoming
PGXN support can handle it gracefully.  For users who want to install it
as an extension, the process is to download source, configure, and
"make extension-install."  Still the extension is not relocatable, users
can specify schema name on CREATE EXTENSION with SCHEMA option.

MADLIB-316

### CODE REVIEW

**Comment by haradh1** [comment]
Not seeing objection, I'm going to merge it today.

**Comment by fschopp** [comment]
Let me take another look. I'll post here once I'm done reviewing.

File: src/madpack/sort-module.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/python
+
+"""
+sort-module.py
+
+Given number of strings, sort them based on the module name which the
+string contains resolving the dependencies.  Example:
```
Comment (fschopp) [inline src/madpack/sort-module.py:7]: I don't understand that comment. Could you rephrase that?

File: src/ports/postgres/CMakeLists.txt [hunk 2]
```diff
@@ -236,6 +236,99 @@ macro(add_current_${PORT_LC}_version)
...
+    # We use X.Y.Z style for version string for it is recommended in PGXN.
+    set(EXTENSION_SQL "${CMAKE_CURRENT_BINARY_DIR}/extension/madlib--${MADLIB_VERSION_MAJOR}.${MADLIB_VERSION_MINOR}.${MADLIB_VERSION_PATCH}.sql")
+
+    # The workhorse for extension target
+    # It is hard to interpret Modules.yml in this cmake file, thus we use
+    # sort-module.py to sort modules based on the dependencies.
+    add_custom_command(OUTPUT ${EXTENSION_SQL}
+        COMMAND cat ${EXTENSION_SQL_BASE}
+            `cd ${CMAKE_BINARY_DIR}/src/madpack &&
+            python sort-module.py ${MODULE_SQL_POST_FILES}` >${EXTENSION_SQL}
```
Comment (fschopp) [inline src/ports/postgres/CMakeLists.txt:301]: The uber-correct way would be using cmake commands like execute_process. I am not sure if this works if the user is using a shell we are not expecting.

**Comment by fschopp** [comment]
Looks fine for now.

File: src/madpack/sort-module.py (hunk 1 above)
Comment (haradh1) [inline src/madpack/sort-module.py:7]: Draft.

<pre>
Sort input strings based on the module name in them such that the module dependencies are resolved.
</pre>

File: src/ports/postgres/CMakeLists.txt (hunk 2 above)
Comment (haradh1) [inline src/ports/postgres/CMakeLists.txt:301]: I tried execute_process to sort the file names, then the dependency order gets broken.  I leave it for now.

**Comment by haradh1** [comment]
Thanks for your inputs!  Merged to master with the feedbacks.
